web/components/rsvp/index.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import { withFirebase } from 'react-redux-firebase';
+import Axios from 'axios';
+import PropTypes from 'prop-types';
+
+import { TextInput } from '../input/text';
+import { Select } from '../input/select';
+import { SecondaryButton, PrimaryButton } from '../input/buttons';
+
+import { tShirtSizes } from './tShirtSizes';
+import { dietRestrictions } from './dietRestrictions';
+
+import praise from '../../assets/emoji/praise.svg';
+import sparkle from '../../assets/emoji/sparkle.svg';
+import celebrate from '../../assets/emoji/celebrate.svg';
+
+import rsvpSuccessGraphic from '../../assets/rsvpSuccess.svg';
+
+class RSVP extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      isIDValid: true,
+      isSubmitted: false,
+      isSubmitting: false,
+      submitFailed: false,
+      emergencyContactName: '',
+      emergencyContactNumber: '',
+      tShirtSize: '',
+      dietRestriction: '',
+      wouldLikeToSee: '',
+    };
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    const { params } = match;
+    const { id } = params;
+    const { firebase } = this.props;
+
+    Axios.post(
+      firebase.nwUtils.getFunctionUrl('checkValidID'),
+      { id },
+      { headers: { 'Content-Type': 'text/plain' } }
+    ).then((res) => {
+      if (res.status === 200) {
+        const { accepted, rsvped, firstName } = res.data;
+
+        if (!accepted) {
+          console.log('Did not find valid ID');
+          this.setState({ isIDValid: false });
+        } else if (rsvped) {
+          console.log('already RSVPed');
+          this.setState({ isSubmitted: true });
+        } else {
+          console.log('Found ID: ' + firstName);
+          // this.setState({ isIDValid: true });
```

Comment:
Can we get rid of any old commented out code?

Reply:
Yep, and got rid of all the unused onblur/error code.